pkg/cli/idle/idle.go

```diff
@@ -427,7 +430,9 @@ func findScalableResourcesForEndpoints(endpoints *corev1.Endpoints, getPod func(
 	controllerRefs := make(map[namespacedCrossGroupObjectReference]struct{})
 	for controllerRef := range immediateControllerRefs {
 		controller, err := getController(controllerRef)
-		if err != nil && errors.IsNotFound(err) {
+		if err != nil {
+			return nil, err
+		} else if errors.IsNotFound(err) {
 			return nil, fmt.Errorf("unable to load %s %q: %v", controllerRef.Kind, controllerRef.Name, err)
 		}
```

Comment:
@miheer, after your change, the `else` branch is unreachable, isn't it?  

Anyway, it seems the logic was broken by an earlier change: <https://github.com/openshift/origin/pull/22919/files#diff-037051bf889204d52030b00c7a2754f939bb8653b22842a3dc53db1259f433d6L420-R420>.  The original intention was any errors _except_ not-found errors should be reported.
```suggestion
		if err != nil && !errors.IsNotFound(err) {
			 if  {
				return nil, fmt.Errorf("unable to load %s %q: %v", controllerRef.Kind, controllerRef.Name, err)
			}
			return nil, err
		}
```
@mfojtik, can you double-check that my analysis of your change and suggestion here is correct?